bridge_adaptivity/module/policies/policy_engine_grade.py

```diff
@@ -0,0 +1,42 @@
+# coding: utf-8
+from bridge_lti.outcomes import update_lms_grades
+from module.engines.engine_vpal import EngineVPAL
+from .base import BaseGradingPolicy
+
+
+class EngineGradedGradingPolicy(BaseGradingPolicy):
+    """Grading policy class calculate grade based upon response from the engine."""
+
+    public_name = 'Engine Graded'
+    require = {
+        'engine': (EngineVPAL,),
+    }
+
+    def _get_points_earned_trials_count(self):
+        """Get points earned and trials count from the sequence.
+
+        :return tuple([trials_count, points_earned])
+        """
+        pass
+
+    def _calculate(self):
+        """
+        Send request to engine and get response with grade.
+
+        :return: received grade from engine.
+        """
+        return self.sequence.group.engine_driver.get_grade(self.sequence)
+
+    @classmethod
+    def get_form_class(cls):
+        from module.forms import BaseGradingPolicyForm
+        return BaseGradingPolicyForm
+
+    def send_grade(self):
+        """Send grade to LMS system.
+
+        Call update_lms_grades(self.context['request'], sequence=self.sequence, user_id=self.context['user_id'])
+        :return: nothing.
```

Comment:
L38 - 39 could be omitted they repeated method's body.

And instead of using code duplication I would rewrite send_grade in the base class as function with the parameter `with_request`
```
def _send_grade(self, with_request=False):
     is_send = with_request and self.context.get('request') or not self.context.get('request')
     if is_send:
         update_lms_grades(self.context.get('request'), sequence=self.sequence, user_id=self.context['user_id'])

@property
def send_grade(self):
     return _send_grade(self)
```

And use this property with the appropriate param in all policies.

Reply:
I can't understand this line 
```
     is_send = with_request and self.context.get('request') or not self.context.get('request')
```

so I changed it to 
```
def _send_grade(self, with_request=False):
    if with_request:
        update_lms_grades(self.context.get('request'), sequence=self.sequence, user_id=self.context['user_id'])
    else:
        update_lms_grades(sequence=self.sequence, user_id=self.context['user_id'])
```